### PR TITLE
[PyCDE] Wrap all CIRCT ops in PyCDE Value veneer.

### PR DESCRIPTION
**Pull request by mikeurbach**
This adds new `pycde.dialects` modules which can be imported instead
of their corresponding `circt.dialects` modules. All ops in the
`pycde.dialects` modules are enhanced to have their `create` method
return a PyCDE Value. The `create` methods already work with PyCDE
Values as operands, because they generally call `support.get_value`,
which supports PyCDE Values through duck typing. Together, this means
that all PyCDE generators can uniformly work with Values, as long as
users stick to the prescribed `create` APIs.

### CODE REVIEW

**Comment by teqdruid** [comment]
Nice! How difficult would it be to use some python magic to create wrapper classes automatically so users don’t have to remember to use the create method? The wrapper constructor would just call it automatically.

**Comment by mikeurbach** [comment]
> create wrapper classes automatically so users don’t have to remember to use the create method

I think it's possible, I can add that too.

**Comment by mikeurbach** [comment]
@teqdruid I am coming back to this. I want to make sure I understand what you're asking for.

> use some python magic to create wrapper classes automatically so users don’t have to remember to use the create method? The wrapper constructor would just call it automatically.

So you want someone to be able to say `AddOp(lhs, rhs)` instead of `AddOp.create(lhs, rhs)`? This means that the wrapped `AddOp` is actually going to be a subclass of `Value`, not `OpView`, as it is now. That should be do-able, I just want to make sure that's correct.

**Comment by teqdruid** [comment]
Yeah, that sounds right

**Comment by mikeurbach** [comment]
@teqdruid let me know what you think of the current revision. Now when you import `pycde.dialects.comb`, you get things like `AddOp` that you can just construct directly, which delegate under the hood to the appropriate `create` call, and wrap the result in the appropriate `Value` class. There are adaptors for things like `__getitem__` so the wrapped class acts like the appropriate `Value` subclass.

**Review by teqdruid — APPROVED** [review]
Nice!

File: frontends/PyCDE/src/pycde/value.py [hunk 1]
```diff
@@ -145,16 +153,71 @@ def __getitem__(self, sub):
...
+          created = self._opview_cls.create(*args, **kwargs)
+          if isinstance(created, support.NamedValueOpView):
+            created = created.opview
+
+          assert len(created.results) == 1
+          value = created.results[0]
+          type = PyCDEType(value.type)
+
+          if isinstance(type.strip, hw.ArrayType):
+            self._value_cls = ListValue
```
[the remark this answers is not in the document]
Comment (mikeurbach) [reply frontends/PyCDE/src/pycde/value.py:201]: originally i was trying to call the appropriate sub-class constructor here and be done with it, but that broke down with the `__getitem__` stuff... i can do it the way you suggested, that is a little cleaner.

**Pull request merged**